app/Services/GraphQL.php

```diff
@@ -15,14 +16,50 @@ public function __construct()
     }
 
     /**
-     * Run a GraphQL query using the client.
+     * Run a GraphQL query using the client and return the data result.
      *
      * @param  $query     String
      * @param  $variables Array
-     * @return \Softonic\GraphQL\Response
+     * @return array|null
      */
     public function query($query, $variables)
     {
-        return $this->client->query($query, $variables);
+        // Use try/catch to avoid any GraphQL related errors breaking the application.
+        try {
+            $response = $this->client->query($query, $variables);
+        } catch (\Exception $exception) {
```

Comment:
Should we swap this to a more specific type of exception? Would we want a network failure, for example, to continue as a real `RuntimeException` (so if this is happening in a queue runner, we could then retry that job later)?

Reply:
Hmm so the graphql-client's `query` method specifies two potential Exceptions in its docblock:
The [`\RuntimeException`](https://github.com/softonic/graphql-client/blob/f60c0ac3945b7413a4cea2cd6cf4894ad3ba788c/src/Client.php#L21) which it sounds like we *don't* want to catch (which makes sense! I didn't realize we could just retry these failed jobs, and I think that sounds like the better option than just logging the error (and sending c.io `null` campaign `title` & `slug` values in the case of the current overarching work)).

Then there's the `\UnexpectedValueException` for an invalid JSON response body. 
https://github.com/softonic/graphql-client/blob/f60c0ac3945b7413a4cea2cd6cf4894ad3ba788c/src/Client.php#L20

I'm thinking now that maybe we don't need to `catch` even the latter for the same reason we won't `catch` the former? I guess I'm not sure of the scenario where we'd even get this `UnexpectedValueException` as opposed to just the `TransferException` (my futzing about with graphql to prompt errors in the response only managed to yield the `TransferException` due to `400`s), but either way, would it make sense to just rely on the retry in both cases?

Reply:
Yeah, I think that's the right call not to catch these. Thanks for looking into it! 🙇

Reply:
And right back atcha -- thanks for all the helpful feedback! (I just had one more outstanding question about the response parsing https://github.com/DoSomething/rogue/pull/889#discussion_r297408575 if you're not tired of me yet 🕺)